kv/index_iter.go

```diff
@@ -122,19 +121,19 @@ type kvIndex struct {
 	indexName string
 	indexID   int64
 	unique    bool
-	prefix    string
+	prefix    Key
 }
 
 // GenIndexPrefix generates the index prefix.
-func GenIndexPrefix(indexPrefix string, indexID int64) string {
+func GenIndexPrefix(indexPrefix Key, indexID int64) Key {
 	buf := make([]byte, 0, len(indexPrefix)+8)
 	buf = append(buf, indexPrefix...)
 	buf = codec.EncodeInt(buf, indexID)
-	return string(buf)
+	return buf
 }
 
 // NewKVIndex builds a new kvIndex object.
-func NewKVIndex(indexPrefix string, indexName string, indexID int64, unique bool) Index {
+func NewKVIndex(indexPrefix Key, indexName string, indexID int64, unique bool) Index {
```

Comment:
I have changed all key prefixes from []byte to Key. I thought it is more clear since 1) Key prefixes are encoded and memory comparable just like keys. 2) When we build an index, prefix is appended directly rather than using codec to encode which is different from other types. 3) Sometimes key prefixes are passed to the `Seek` method just like keys.